decimal.go

```diff
@@ -35,13 +35,17 @@ type Decimal struct {
 }
 
 // New creates a new decimal with the given coefficient and exponent.
+//
+// REVIEW: I wonder if a Make function would also be useful. Ideally you wouldn't
+// always need to allocate the outer struct onto the heap.
```

Comment:
Ah. I'd like to add this in the future when we find a place where it is clearly beneficial.